test/classification/training/test_trainer.py

```diff
@@ -216,6 +216,35 @@ def test_warmup(self):
         trainer.fit(model, [dataloaders[0]])
         self.assertEqual(trainer.warmup_steps, 1)
 
+    def test_save_load(self):
+        fd, checkpoint_path = tempfile.mkstemp()
+        non_base_config = {"n_epochs": 2, "progress_bar": False}
+
+        trainer1 = Trainer(**base_config, lr_scheduler="exponential")
+        trainer1.fit(model, [dataloaders[0]])
+
+        trainer2 = Trainer(**non_base_config, lr_scheduler="linear")
+
+        trainer1.save(checkpoint_path)
+        trainer2.load(checkpoint_path, model=model)
+        self.assertEqual(trainer1.config, trainer2.config)
+
+        self.assertEqual(
+            trainer1.optimizer.state_dict()["param_groups"],
+            trainer2.optimizer.state_dict()["param_groups"],
+        )
+
+        for k in trainer1.optimizer.state_dict()["state"].keys():
+            print()
+            self.assertTrue(
+                torch.eq(
+                    trainer1.optimizer.state_dict()["state"][k]["exp_avg"],
+                    trainer2.optimizer.state_dict()["state"][k]["exp_avg"],
```

Comment:
why are we only checking equivalence of these fields? could we check the entire `state_dict`'s values?